dagjose/jws_signature_assembler.go

```diff
@@ -90,7 +91,7 @@ func (j *jwsSignatureAssembler) AssignLink(l ipld.Link) error {
 	return jwsSignatureMixin.AssignLink(l)
 }
 func (j *jwsSignatureAssembler) AssignNode(n ipld.Node) error {
-	return fmt.Errorf("not implemented")
+	return datamodel.Copy(n, j)
```

Comment:
Not sure if this and any of the other `datamodel.Copy` changes are necessary. I haven't run into any issues in my testing, but maybe things are different in `go-ipld-prime v0.12.0`.

Reply:
The reason for the `datamodel.Copy` is not because the CBOR encoding won't work without it, it is to make sure that only _valid_ JOSE gets CBOR encoded.

Without this change I'm able to store any valid CBOR object even if it is _invalid_ JOSE.